### ISSUE TITLE
First clip doesn't load

### ISSUE THREAD

**Issue by woempiej** [dermotduffy/frigate-hass-card#1368]
Hi,

I'm using the Frigate Lovelace card. Since the latest update the Lovelace card doesn't show the 1st (latest) video anymore.
I'm using the card in the "most recent clip" view.

![afbeelding](https://github.com/dermotduffy/frigate-hass-card/assets/62903344/0f12c1a6-d636-4918-acf5-d861f104a9d7)

I need to
Android HA app:
Swipe to the previous video and then swipe back to the newest video to be able to watch the newest video.
Firefox browser:
Navigate to another dashboard tab and go back to my "security" tab to be able to watch the newest video.
or
Choose to edit the HA dashboard > Then it showes the video correctly
Chrome browser:
Navigate to another dashboard tab and go back to my "security" tab to be able to watch the newest video.
or
Choose to edit the HA dashboard > Then it showes the video correctly

I did think about the browser cache and history wipe.

No error(s) logs in the HA log
No error(s) in the Frigate log

Yesterday I rebuild all of the Frigate installation:
Deleted HA add-on
Deleted HA integration > Wich the Lovelace card is comming from
Deleted all Frigate related files via the file editor
Deleted all Media
Media > Frigate (Integration folder)
Media > My Media > frigate (Add-On folder)

Deleted the Lovelace card and builded it back up via the GUI

```yaml
  - type: custom:frigate-card
    cameras:
      - camera_entity: camera.frigate_cam1
      - camera_entity: camera.frigate_cam2
      - camera_entity: camera.frigate_cam3
    view:
      default: clip
      update_force: true
      scan:
        show_trigger_status: true
    menu:
      style: outside
      buttons:
        media_player:
          enabled: false
        timeline:
          enabled: false
        live:
          enabled: false
        snapshots:
          enabled: false
    media_viewer:
      controls:
        title:
          mode: none
        thumbnails:
          mode: none
        timeline:
          mode: below
          window_seconds: 28800
    media_gallery:
      controls:
        thumbnails:
          show_favorite_control: false
          show_timeline_control: false
          show_download_control: false
        filter:
          mode: none
```

**Comment by dermotduffy** [comment]
Just to make sure I understand, the most recent clip doesn't work, but all the others do? Is this the case no matter how recently the clip was? i.e. even if it was hours ago, the first clip will never work?

**Comment by woempiej** [comment]
Hi,

That is correct.
I have 3 camera's configured in the card. Only the first (default) camera clip acts this way.

**Comment by dermotduffy** [comment]
> I have 3 camera's configured in the card. Only the first (default) camera clip acts this way.

And can you view the same clip that won't load via the HA media browser? (i.e. outside the card, but inside the HomeAssistant `Media` sidebar tab)

**Comment by woempiej** [comment]
In the media browser I see the same issue. The first clip doesn't load

**Comment by wizu** [comment]
I have same problem.
In browser console, I have an error, when click on first clip:

```
96996.w7vKRby48d0.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'KeySystemFormats')
    at 96996 (96996.w7vKRby48d0.js:1:283711)
    at Function.c (bootstrap:19:32)
    at async p.value (ha-hls-player.ts:112:11)
96996 @ 96996.w7vKRby48d0.js:1
c @ bootstrap:19
log-message.ts:37 
        
GET https://xxx/unknown/webpack/bootstrap 404 (Not Found)
```

**Comment by woempiej** [comment]
Hi dermotduffy.

I did report it upstream first ;)
https://github.com/blakeblackshear/frigate/issues/9827

I think it is not a Frigate Add-on media browser problem because in the media browser of the add-on I do not have issues.

I have 2 media browser directories:
1. From the Frigate Add-on: Media > My Media > frigate
2. From this Frigate integration: Media > Frigate

Media directory number 1 does not show this behaviour. Some clips are split in 2 peaces inside the directory.
The most recent clip (recording) of Media directory number 2 is showing the same behaviour as the Lovelace card.

**Comment by woempiej** [comment]
![afbeelding](https://github.com/dermotduffy/frigate-hass-card/assets/62903344/ec18cfe8-64f0-48ea-bd32-ffbae25787e4)

Frigate Add-on 0.13.1

**Comment by dermotduffy** [comment]
I'm sorry you are being bounced around! On the other issue you mentioned the card and not that this ALSO doesn't work in the stock HA media browser without the card being involved.

> I think it is not a Frigate Add-on media browser problem because in the media browser of the add-on I do not have issues.

I think the issue can only be with one of these (none of which are the Frigate add-on, and none of which are the card):

* Frigate itself ... which passes the video to ...
* ... the Frigate integration ... which plays through ...
* ... the video player that resides in the HASS frontend to play videos

Put simply: You could uninstall the card entirely, and it still wouldn't play in the HA media browser. Can't be the card ;-)

**Comment by woempiej** [comment]
Hi,

Correct me if I am wrong but your integration provides a media directory.
In that directory I can reproduce the issue. In the Frigate Add-on directory I can't

![afbeelding](https://github.com/dermotduffy/frigate-hass-card/assets/62903344/e6b1b345-cea6-4edf-b87d-3585708f4f96)

**Comment by dermotduffy** [comment]
> Correct me if I am wrong but your integration is creating a media directory. In this directory I can reproduce the issue.

That's the [integration](https://github.com/blakeblackshear/frigate-hass-integration), this repo is the card.

**Comment by NickM-27** [comment]
> Swipe to the previous video and then swipe back to the newest video to be able to watch the newest video.

This indicates that it is not a backend issue right? Because the same request works after updating the ui?

**Comment by dermotduffy** [comment]
> This indicates that it is not a backend issue right? Because the same request works after updating the ui?

I missed that, great point! However, there's must be more than just the card going on here since the issue happens in the stock media browser without the card at all.

@woempiej  I wonder if you try to open it in the HA media browser (and it fails), but then you try to open it a second time, does it succeed? [i.e. are we seeing the same symptoms in the media browser as in the card].

Both the card and HA share the same video player code, so it could be that. Or I guess it could be that somehow something on the backend is returning different data on the first and subsequent requests?

**Comment by woempiej** [comment]
Hi,

Here a small tutorial of my actions to try to reproduce

![afbeelding](https://github.com/dermotduffy/frigate-hass-card/assets/62903344/d748f5f6-b4d8-4623-9182-97f91f661f1e)

@dermotduffy Yes the second time in the media browser there is no issue.
Because of this, I thought it was a problem just with the default camera in the card but it is not!
When I log out of HA en close the browser > Login again ant follow the above tutorial to another camera > It is showing the same behaviour > Latest recording isnt't playing

**Comment by dermotduffy** [comment]
How strange. Thanks for the walk-thru screenshot. For the record, I can do the same, go to the media browser, choose Recordings, choose a camera, choose the latest available hour -- it will play first time every time. I'm on HA 2024.1.3 (slightly  older than you), I can try to upgrade later to see if perhaps something changed in HA itself?

Suggestion: In one window, navigate to "Logs" in your Frigate instance, and choose "Nginx" on the top-right. In another window, repeat your steps above (with the first request than fails, and the second that succeeds), then paste in what new log lines are created. Interesting ones likely contain something like "GET /vod/2024-02/..." [these are the requests HA is making to your backend, I'm curious if both requests are being received at all].

**Comment by dermotduffy** [comment]
Looks like both requests are served okay from the backends perspective (both http status `200`), although the response sizes look small compared to my testing, but that could be a red herring.

@woempiej Do you get any messages in your Javascript console (press F12 in Chrome), when you attempt to play the video from the media browser?

**Comment by woempiej** [comment]
![afbeelding](https://github.com/dermotduffy/frigate-hass-card/assets/62903344/220ee282-1e9a-471e-bd34-a41316f4752e)

**Comment by PablaV** [comment]
Anyone have any working solutions? I am experiencing the exact same issue as described above

**Comment by woempiej** [comment]
Hi all,

Issue is fixed in HA 2024.3.0

**Comment by PablaV** [comment]
Can confirm that 2024.3 has fixed this issue
